config/application.rb

```diff
@@ -66,7 +66,8 @@ class Application < Rails::Application
     # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
     # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
     config.i18n.default_locale = ENV["LOCALE"]
-
+    I18n.locale = config.i18n.locale = config.i18n.default_locale
```

Comment:
What I understand is that, for reasons I don't know, delayed_job or something else is not using the variable default_locale that was defined in application.rb but I18n.locale. So it gets initialized with the same value as default_locale which itself is taken from an  export of LOCALE in .bash_rc

At least, it fixed the issue we had  in the french instance.

-----Message d'origine-----
De : "Maikel" <notifications@github.com>
Envoyé : ‎08/‎01/‎2016 08:27
À : "openfoodfoundation/openfoodnetwork" <openfoodnetwork@noreply.github.com>
Cc : "marito59" <maritore59@yahoo.fr>
Objet : Re: [openfoodnetwork] Fix 2 localisation issues (#770)

In config/application.rb:
> @@ -66,7 +66,8 @@ class Application < Rails::Application
>      # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
>      # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
>      config.i18n.default_locale = ENV["LOCALE"]
> -
> +    I18n.locale = config.i18n.locale = config.i18n.default_locale
Thanks, interesting. Do you know if I18n.locale or config.i18n.locale must be set? I'm wondering which initialisation is missing in delayed_job so that it doesn't pick up the default locale.
—
Reply to this email directly or view it on GitHub.